thirdeye-ui/src/app/components/impact-dashboard/detection-performance/detection-performance.component.tsx

```diff
@@ -0,0 +1,180 @@
+ * License is distributed on an "AS IS" BASIS, WITHOUT * WARRANTIES OF ANY KIND,
+ * either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+import React, { ReactElement, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+
+// Styles
+import { useDetectionPerformanceStyles } from "./detection-performance.styles";
+
+// Components
+import { AnalysisPeriod } from "../common/anaylysis-period/analysis-period.component";
+import { LineGraph } from "./line-graph.component";
+import { WeeklyGraph } from "./weekly-graph.component";
+import { SectionHeader } from "../common/section-header/section-header.component";
+import { MenuItem, Select } from "@material-ui/core";
+
+// Interfaces
+import { DetectionPerformanceProps } from "./detection-performance.interfaces";
+
+// APIs
+import { useApiRequests } from "./api";
+
+// Data
+import { useGetData } from "./data";
+
+const defaultAlertDropdownOption = { id: -1, name: "All alerts" };
+
+export const DetectionPerformance = ({
+    anomalies: allAnomalies,
+    previousPeriodAnomalies: allPreviousPeriodAnomalies,
+    analysisPeriods,
+    selectedAnalysisPeriod,
+    onAnalysisPeriodChange,
+}: DetectionPerformanceProps): ReactElement => {
+    const { t } = useTranslation();
+    const [alertDropdownOptions, setAlertDropdownOptions] =
+        useState<{ id: number; name: string }[]>();
+    const [selectedAlert, setSelectedAlert] = useState<{
+        id: number;
+        name: string;
+    }>(defaultAlertDropdownOption);
+
+    const {
+        alerts,
+        currentPeriodAnomaliesByAlert,
+        previousPeriodAnomaliesByAlert,
+    } = useApiRequests({
+        selectedAlert,
+        defaultAlertDropdownOption,
+        selectedAnalysisPeriod,
+    });
+    const { anomalies, previousPeriodAnomalies } = useGetData({
+        selectedAlert,
+        defaultAlertDropdownOption,
+        allAnomalies,
+        allPreviousPeriodAnomalies,
+        currentPeriodAnomaliesByAlert,
+        previousPeriodAnomaliesByAlert,
+    });
+
+    const componentStyles = useDetectionPerformanceStyles();
+
+    useEffect(() => {
+        const options =
+            alerts?.map((alert) => {
+                return { id: alert.id, name: alert.name };
+            }) || [];
+        setAlertDropdownOptions([...options, defaultAlertDropdownOption]);
+    }, [alerts]);
+
+    const handleAlertChange = (
+        event: React.ChangeEvent<{ value: unknown }>
+    ): void => {
+        alertDropdownOptions &&
+            setSelectedAlert(
+                alertDropdownOptions.find(
+                    (alertDropdownOption) =>
+                        alertDropdownOption.id === event.target.value
+                )!
+            );
+    };
+
+    return (
+        <>
+            <div className={componentStyles.sectionHeading}>
```

Comment:
Why not use `Box` instead of `div`?

Reply:
Box renders as div only and then I will anyways have to apply the style, so thats an extra import